core/src/main/web/app/mainapp/mainapp.js

```diff
@@ -541,8 +541,53 @@
             }
           };
 
+          function _closeNotebook() {
+            var self = this;
+            var closeSession = function() {
+              bkSessionManager.close().then(function() {
+                bkCoreManager.gotoControlPanel();
+              });
+            };
+            if (bkSessionManager.isNotebookModelEdited() === false) {
+              closeSession();
```

Comment:
this is a possible race condition since we are not handling the result of the promise